### PR TITLE
feat: Change Staker to Use Primitive Parameter Types

### PR DESCRIPTION
**Pull request by notJoon**
# Description

part of #12

### CODE REVIEW

File: staker/staker.gno [hunk 1]
```diff
@@ -43,45 +43,52 @@ func Init(token grc20.Token) {
 }
 
 // StakeAmount stakes the specified amount for the given delegate
-func StakeAmount(delegate std.Address, amount *u256.Uint) error {
+func StakeAmount(delegate string, amount uint64) error {
```
Comment (r3v4s) [inline staker/staker.gno]: Previous code was using u256 for amount type. Which means number can be exceed default uint64 type.

However, it's currently using uint64.

If uint64 is enough type, resolve this conversation.
If not, use string for amount then convert to u256.

File: staker/staker.gno [hunk 2]
```diff
@@ -43,45 +43,52 @@ func Init(token grc20.Token) {
...
+	delegateAddr := std.Address(delegate)
```
Comment (r3v4s) [inline staker/staker.gno:49]: Check if delegate is valid address

File: staker/staker.gno [hunk 3]
```diff
@@ -208,14 +219,14 @@ func GetAverageDelegated(delegate std.Address, start, end uint64) (*u256.Uint, e
 }
 
 // GetAverageDelegatedOverLast gets the average amount delegated over the last period seconds.
-func GetAverageDelegatedOverLast(delegate std.Address, period uint64) (*u256.Uint, error) {
+func GetAverageDelegatedOverLast(delegate string, period uint64) (*u256.Uint, error) {
```
Comment (r3v4s) [inline staker/staker.gno]:
```suggestion
func GetAverageDelegatedOverLast(delegate string, period uint64) (string, error) {
```

FYI, gno http rpc service can not parse pointer return value.
How about we return string??

File: staker/staker_test.gno [hunk 4]
```diff
@@ -157,38 +158,21 @@ func TestWithdrawAmount(t *testing.T) {
...
-}
-
 func TestGetAverageDelegated(t *testing.T) {
 	token := newMockGRC20("Test Token", "TST", 42)
 	Init(token)
 
-	delegate := std.Address("delegate")
-	stakeAmount := u256.NewUint(0)
+	delegate := "delegate"
+	delegateAddr := std.Address(delegate)
```
Comment (r3v4s) [inline staker/staker_test.gno]: testutils to generate valid address

**Review by r3v4s — CHANGES_REQUESTED** [review]
few things to change

File: staker/staker_test.gno (hunk 4 above)
Comment (notJoon) [reply staker/staker_test.gno]: fixed https://github.com/gnoswap-labs/governance/pull/14/commits/99ef445de0ad28686998bd36ff249a3e9dc686c1

File: staker/staker.gno (hunk 3 above)
Comment (notJoon) [reply staker/staker.gno]: changed all the functions to returning string types
https://github.com/gnoswap-labs/governance/pull/14/commits/6fd92005e9110f1ba48fbcf35e2d2eb2bbd19aa0

File: staker/staker.gno (hunk 1 above)
Comment (notJoon) [reply staker/staker.gno]: amount param type changed: uint64 -> string

https://github.com/gnoswap-labs/governance/commit/6fd92005e9110f1ba48fbcf35e2d2eb2bbd19aa0

File: staker/staker.gno (hunk 2 above)
Comment (notJoon) [reply staker/staker.gno:49]: I'm not sure how to check valid address propoerly, So I just comparing is this empty string or not.

https://github.com/gnoswap-labs/governance/pull/14/commits/37845cc98a4cf14e140259bd291469192121dce6

Comment (r3v4s) [reply staker/staker.gno:49]: gno has IsValid for address
https://docs.gno.land/reference/stdlibs/std/address#isvalid

Comment (notJoon) [reply staker/staker.gno:49]: thanks! fixed

https://github.com/gnoswap-labs/governance/pull/14/commits/348d47435f21d5b7150bfee0d5bf56089a7eb12c

**Review by dongwon8247 — APPROVED** [review]

**Review by r3v4s — APPROVED** [review]

**Pull request merged**